### PR TITLE
Continue polling when the last poll returned more jobs that the lowest priority request

### PR DESCRIPTION
**Pull request by oeoeaio**
Resolves https://github.com/que-rb/que/issues/345.

Alternative to https://github.com/que-rb/que/pull/348

### CODE REVIEW

**Review by ZimbiX — CHANGES_REQUESTED** [review]
See https://github.com/que-rb/que/pull/348#discussion_r819217705

File: lib/que/poller.rb [hunk 1]
```diff
@@ -265,5 +263,12 @@ def cleanup(connection)
         SQL
       end
     end
+
+    private
+
+    def poll_satisfied?(priorities, jobs)
+      lowest_priority = priorities.keys.max
```
Comment (oeoeaio) [inline lib/que/poller.rb:270]: This was just a first crack at this, can probably do something a little more robust to select the requests for the lowest priority workers…

Comment (ZimbiX) [reply lib/que/poller.rb:270]: What's not robust about it? Looks like it's always an integer - [`any` has been replaced by nil](https://github.com/que-rb/que/blob/e51f6c1826f12ccf07018b1cee3c9adfb456b3f7/bin/command_line_interface.rb#L72-L73) and then [nil replaced by the priority maximum](https://github.com/que-rb/que/blob/e51f6c1826f12ccf07018b1cee3c9adfb456b3f7/lib/que/job_buffer.rb#L128).

Comment (oeoeaio) [reply lib/que/poller.rb:270]: Oh! I missed that. It’s probably fine then.

**Review by ZimbiX — APPROVED** [review]
Yeah, let's go with this over https://github.com/que-rb/que/pull/348. Simple's good

**Pull request merged**